105-strimzi-drain-cleaner-namespace-filtering-proposal.md

```diff
@@ -0,0 +1,269 @@
+
+### Integration Tests
+- Deploy drain cleaner with namespace filtering in multi-namespace test environment
+- Verify eviction handling for both watched and unwatched namespaces
+- Confirm no API calls made to unwatched namespaces
+- Validate log output and behavior
+
+### Backward Compatibility Tests
+- Verify existing deployments continue working without configuration changes
+- Test upgrade scenarios from previous versions
+- Confirm default behavior matches current implementation
+
+## Graduation/Rollout Plan
+
+### Phase 1: Implementation and Testing
+- Implement core functionality in drain cleaner
+- Add comprehensive test coverage
+- Update packaging configurations
+- Documentation updates
+
+### Phase 2: Beta Release
+- Include in next minor Strimzi release (0.48.0+)
+- Mark as beta feature in release notes
+- Gather community feedback
+- Monitor for issues in test environments
+
+### Phase 3: General Availability
+- Promote to stable feature status
+- Include in documentation as recommended pattern for multi-tenant environments
+- Consider for inclusion in Helm chart best practices
+
+## Documentation Updates
+
+- **Configuration Reference**: Add `STRIMZI_DRAIN_WATCH_NAMESPACES` to environment variables documentation
+- **Multi-Tenant Guide**: Create section on namespace filtering for enterprise deployments
+- **Security Guide**: Document RBAC considerations and filtering benefits
+- **Troubleshooting**: Add guidance for configuration and debugging
+- **Migration Guide**: Document upgrade considerations for existing deployments
+
+## Metrics and Monitoring
+
+### Success Metrics
+- Reduction in permission error log entries
+- Decreased API server load from drain cleaner
+- Community adoption in multi-tenant environments
+- User feedback on deployment simplification
+
+### Monitoring Considerations
+- Log volume reduction in restricted environments
+- API call patterns to Kubernetes API server
+- Webhook response times and success rates
+
+## Future Enhancements
+
+While not part of this proposal, future work could include:
+
+1. **Dynamic Configuration**: Support for runtime configuration updates
+2. **Namespace Pattern Matching**: Support for regex or wildcard patterns
+3. **Enhanced Logging**: Structured logging with namespace filtering statistics
+4. **Metrics Exposure**: Prometheus metrics for namespace filtering behavior
+
+## References
+
+- [Strimzi Drain Cleaner Documentation](https://strimzi.io/docs/operators/latest/deploying.html#drain-cleaner)
+- [Multi-Tenant Kubernetes Best Practices](https://kubernetes.io/docs/concepts/policy/pod-security-policy/)
+- [Kubernetes Admission Controllers](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/)
+- [Related Pull Request: #164](https://github.com/strimzi/drain-cleaner/pull/164)
+
+---
+
+**Proposal Author**: Roman Melnyk (aywengo@gmail.com)  
+**Implementation Reference**: [Drain Cleaner PR #164](https://github.com/strimzi/drain-cleaner/pull/164) 
```

Comment:
This is not really needed either. Feel free to use the links in the text where/if needed.

Reply:
Removed